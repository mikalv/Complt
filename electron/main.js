const menubar = require('menubar');

const mb = menubar({
  height: 600,
  tooltip: 'Complt',
  showDockIcon: true,
});

mb.on('ready', () => {
  console.log('app is ready'); // eslint-disable-line no-console
});
