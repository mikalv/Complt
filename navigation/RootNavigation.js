import React from 'react';
import {
  Text,
  View,
  StyleSheet,
} from 'react-native';
import {
  StackNavigation,
  DrawerNavigation,
  DrawerNavigationItem,
} from '@exponent/ex-navigation';
import {
  MaterialIcons,
} from 'react-native-vector-icons';
import colors from '../constants/colors';

const styles = StyleSheet.create({
  header: {
    height: 180,
    width: 300,
    backgroundColor: colors.drawerHeader,
  },
  buttonTitleText: {
    color: '#222',
    marginLeft: 18,
    fontSize: 17,
    marginTop: 2,
    marginBottom: 2,
  },
  icon: {
    color: '#999',
    fontSize: 20,
    marginTop: 2,
    marginBottom: 2,
  },
  selectedText: {
    color: colors.primary,
  },
  selectedItemStyle: {
    backgroundColor: '#E8E8E8',
  },
});

const renderIcon = (icon, isSelected) => (
  <MaterialIcons
    style={[styles.icon, isSelected ? styles.selectedText : null]}
    name={icon}
  />
);

const renderTitle = (title, isSelected) => (
  <Text
    style={[styles.buttonTitleText, isSelected ? styles.selectedText : null]}
  >{title}</Text>
);


const RootNavigation = () => (
  <DrawerNavigation
    initialItem="inbox"
    drawerWidth={300}
    renderHeader={() =>
      <View style={styles.header} />}
  >
    <DrawerNavigationItem
      id="inbox"
      selectedStyle={styles.selectedItemStyle}
      renderIcon={isSelected => renderIcon('inbox', isSelected)}
      renderTitle={isSelected => renderTitle('Inbox', isSelected)}
    >
      <StackNavigation initialRoute="inbox" />
    </DrawerNavigationItem>

    <DrawerNavigationItem
      id="projects"
      selectedStyle={styles.selectedItemStyle}
      renderIcon={isSelected => renderIcon('home', isSelected)}
      renderTitle={isSelected => renderTitle('Projects', isSelected)}
    >
      <StackNavigation initialRoute="projects" />
    </DrawerNavigationItem>

    <DrawerNavigationItem
      id="settings"
      selectedStyle={styles.selectedItemStyle}
      renderIcon={isSelected => renderIcon('settings', isSelected)}
      renderTitle={isSelected => renderTitle('Settings', isSelected)}
    >
      <StackNavigation initialRoute="settings" />
    </DrawerNavigationItem>
  </DrawerNavigation>
);

export default RootNavigation;
