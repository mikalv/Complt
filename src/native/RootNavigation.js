import { DrawerNavigator } from 'react-navigation';
import Inbox from './routes/Inbox';
import Projects from './routes/Projects';

export default DrawerNavigator({
  Inbox: { screen: Inbox },
  Projects: { screen: Projects },
});
