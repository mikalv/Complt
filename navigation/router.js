import {
  createRouter,
} from '@exponent/ex-navigation';

import InboxScreen from '../screens/InboxScreen';
import ProjectsScreen from '../screens/ProjectsScreen';
import SettingsScreen from '../screens/SettingsScreen';
import RootNavigation from './RootNavigation';

export default createRouter(() => ({
  inbox: () => InboxScreen,
  projects: () => ProjectsScreen,
  settings: () => SettingsScreen,
  rootNavigation: () => RootNavigation,
}));
