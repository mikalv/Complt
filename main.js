import Exponent from 'exponent';
import React from 'react';
import {
  StatusBar,
  StyleSheet,
  View,
} from 'react-native';
import {
  NavigationProvider,
  StackNavigation,
} from '@exponent/ex-navigation';
import {
  MaterialIcons,
} from '@exponent/vector-icons';
import router from './navigation/router';
import cacheAssetsAsync from './utils/cacheAssetsAsync';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});

class AppContainer extends React.Component {
  state = {
    appIsReady: false,
  }

  componentWillMount() {
    this._loadAssetsAsync();
  }

  async _loadAssetsAsync() {
    try {
      await cacheAssetsAsync({
        images: [
          require('./assets/images/exponent-wordmark.png'),
        ],
        fonts: [
          MaterialIcons
          .font,
          { 'space-mono': require('./assets/fonts/SpaceMono-Regular.ttf') },
        ],
      });
    } catch (e) {
      console.warn( // eslint-disable-line no-console
        'There was an error caching assets (see: main.js), perhaps due to a ' +
        'network timeout, so we skipped caching. Reload the app to try again.',
      );
      console.log(e.message); // eslint-disable-line no-console
    } finally {
      this.setState({ appIsReady: true });
    }
  }

  render() {
    if (this.state.appIsReady) {
      return (
        <View style={styles.container}>
          <NavigationProvider router={router}>
            <StackNavigation id="root" initialRoute={router.getRoute('rootNavigation')} />
          </NavigationProvider>
          <StatusBar barStyle="default" />
        </View>
      );
    }
    return (
      <Exponent.Components.AppLoading />
    );
  }
}

Exponent.registerRootComponent(AppContainer);
