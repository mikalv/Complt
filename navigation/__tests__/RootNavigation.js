import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { NavigationProvider } from '@exponent/ex-navigation';
import RootNavigation, { renderTitle, renderIcon } from '../RootNavigation';
import router from '../router';
import '../../main'; // importing to get coverage correctly

describe('RootNavigation', () => {
  it('renders without crashing', () => {
    renderer.create(<NavigationProvider router={router}><RootNavigation /></NavigationProvider>);
  });
  describe('renderTitle', () => {
    it('renders correctly when it\'s selected', () => {
      const component = renderer.create(renderTitle('Inbox', true));
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders correctly when it\'s not selected', () => {
      const component = renderer.create(renderTitle('Inbox', false));
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('renderIcon', () => {
    it('renders correctly when it\'s selected', () => {
      const component = renderer.create(renderIcon('inbox', true));
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders correctly when it\'s not selected', () => {
      const component = renderer.create(renderIcon('inbox', false));
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
