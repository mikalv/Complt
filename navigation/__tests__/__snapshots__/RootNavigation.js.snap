exports[`RootNavigation renderIcon renders correctly when it's not selected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail" />
`;

exports[`RootNavigation renderIcon renders correctly when it's selected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail" />
`;

exports[`RootNavigation renderTitle renders correctly when it's not selected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#222",
        "fontSize": 17,
        "marginBottom": 2,
        "marginLeft": 18,
        "marginTop": 2,
      },
      null,
    ]
  }>
  Inbox
</Text>
`;

exports[`RootNavigation renderTitle renders correctly when it's selected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#222",
        "fontSize": 17,
        "marginBottom": 2,
        "marginLeft": 18,
        "marginTop": 2,
      },
      Object {
        "color": "#03CEA4",
      },
    ]
  }>
  Inbox
</Text>
`;
